File: Question-Bank-Service/src/main/java/com/example/QuestionBankService/controllers/QuestionController.java
package com.example.QuestionBankService.controllers;

import com.example.QuestionBankService.models.Question;
import com.example.QuestionBankService.services.QuestionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Optional;

@RestController
@RequestMapping("/question")
public class QuestionController {
    @Autowired
    QuestionService service;

    @PostMapping("/create")
    public HttpStatus createQuetions(@RequestBody Question question)
    {
        service.createQuestion(question);
        return HttpStatus.CREATED;
    }
    @GetMapping("/findById/{id}")
    public Optional<Question> getQuestionById(@PathVariable String id)
    {

        return service.getQuetionById(id);
    }
    @GetMapping("/all")
    public List<Question> getAllQuestions() {
        return service.getAllQuetions();
    }

    @DeleteMapping("/delete/{id}")
    public String deleteQuestionById(@PathVariable String id)
    {
        return service.deleteQuestionById(id);
    }
    @DeleteMapping("/delete")
    public String deleteQuestion(@RequestBody Question question)
    {
        return service.deleteQuestion(question);
    }
    @PutMapping("/update")
    public String updateQuestion(@RequestBody Question question)
    {
        return service.updateQuestion(question);
    }
}

File: front-end/src/controllers/question.controller.ts
import { QuestionService } from './../services/question.service';
import { environment } from "src/environments/environment.development";
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Question } from 'src/models/question.model';
var apiUrl = environment.questionApiUrl;
var httpLink = {
  getAllQuestions: apiUrl + "/question/all",
  deleteQuestionById: apiUrl + "/question/delete",
  getQuestionDetailById: apiUrl + "/question/findById",
  saveQuestion: apiUrl + "/question/create",
  updateQuestion:apiUrl+"/question/update"
  }
@Injectable({
    providedIn: 'root'
})
export class QuestionController
{
  constructor(private questionService: QuestionService) { }

  // public getAllQuestions(): Question[] {

  //   return this.questionService.getAllQuestions(httpLink.getAllQuestions);
  //   console.log("HERE");
  // }
  public getAllQuestions(): Observable<any> {
    return this.questionService.get(httpLink.getAllQuestions);
  }
  public deleteQuestionById(id: any): any {
    return this.questionService.delete(httpLink.deleteQuestionById + '/' + id);
  }
  public getQuestionDetailById(model: any): Observable<any> {
    return this.questionService.get(httpLink.getQuestionDetailById + '/' + model);
  }
  public saveQuestion(question: Question): void {
    return this.questionService.addQuestion(httpLink.saveQuestion, question);
  }
  public updateQuestion(question :Question):any{
    return this.questionService.updateQuestion(httpLink.updateQuestion,question);
  }
}
